File: solutions/python/isbn-verifier/1/isbn_verifier.py
def is_valid(isbn):
    isbn = isbn.replace("-", "")
    if len(isbn) != 10:
        return False
    start = 10
    total = 0
    for i, c in enumerate(isbn):
        if not c.isdigit() and c != "X":
            return False
        if c == "X" and i != 9:
            return False
        total += start * (10 if c == "X" else int(c))
        start -= 1

    return total % 11 == 0

File: solutions/typescript/isbn-verifier/1/isbn-verifier.ts
export function isValid(isbn: string): boolean {
  if (isbn.match(/[^0-9X-]/)) return false
  let digits = isbn.replace(/-/g, '').split('')
  if (digits.length !== 10) return false
  if (digits[9] === 'X') digits[9] = '10'
  let sum = 0
  for (let i = 0; i < digits.length; i++) {
    sum += parseInt(digits[i]) * (10 - i)
  }
  if (sum % 11 === 0) return true
  return false
}
